backend/logic/printful/sync/stockUpdate.js

```diff
@@ -0,0 +1,61 @@
+const get = require('lodash/get')
+const { decryptConfig } = require('../../../utils/encryptedConfig')
+
+const { Product } = require('../../../models')
+const { getLogger } = require('../../../utils/logger')
+
+const log = getLogger('utils.printful.writeProductData')
+
+/**
+ * Appends a productId to a collection.
+ * And creates collection if it doesn't exist
+ *
+ * @param {Object} shop
+ * @param {Number|String} productId
+ * @param {Array<Object>} variants
+ */
+const stockUpdate = async (shop, productId, variants) => {
+  try {
+    const shopConfig = decryptConfig(shop.config)
+    const shopInventory = get(shopConfig, 'inventory')
+    if (shopInventory) {
+      // Add to products model
+      const variantsStock = variants.reduce(
+        (stockData, variant) => ({
+          ...stockData,
+          [variant]: 0
+        }),
+        {}
+      )
+
+      const updatedStockData = {
+        productId: productId.toString(),
+        shopId: shop.id,
+        stockLeft: 0,
```

Comment:
My understanding is that stockLeft is the cumulative stock left across all the variants.
When this method gets called is stockLeft always completely depleted? Or does it gets calls every time a variant's stock changes?

Reply:
> When this method gets called is stockLeft always completely depleted? Or does it gets calls every time a variant's stock changes?

It only sends the out of stock and discontinued variants. It does not send the remaining stock for other variants. It could be a problem if only some variants are out of stock. I can compare variants array to find in stock items and set the stock to a higher number like 10000.

Reply:
Updated